Modules/Module_Processing/xASL_init_DefaultEffectiveResolution.m

```diff
@@ -30,15 +30,23 @@
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
 % REFERENCES: Petr, 2018 MAGMA; Vidorreta 2013 Neuroimage
 %
-% Copyright 2015-2020 ExploreASL
+% Copyright 2015-2024 ExploreASL
 
 %% ----------------------------------------------------------------------------------------
 %% Admin
-tIM = xASL_io_ReadNifti(PathASL);
+[tIM, tJSON] = xASL_io_ReadNifti(PathASL);
 NativeResolution = tIM.hdr.pixdim(2:4);
-if ~isfield(x.Q,'Sequence')
-    warning(['Setting x.Q.Sequence missing, skipping']);
+
+% Obtain the x.Q.Sequence field from the json
+if ~isempty(tJson) && isfield(tJson, 'Q') && isfield(tJson.Q, 'Sequence')
```

Comment:
We still need to run xASL_bids_parms2BIDS here right, in case we're loading a BIDS json?

Reply:
Haha. We don't. Cause BIDS2legacy conversion is an integral part of io_readNifti :) So bids_parms2BIDS is called inside the JSON loading.